metrics/web/docs/features/remote_config/01_remote_config_design.md

```diff
@@ -0,0 +1,74 @@
+5. Open the appeared dropdown and enable the following APIs: `Firebase Installations API` and `Firebase Remote Config API`.
+
+Once the above steps are completed, the `Remote Config API` is enabled for your web project. Please note that it may take some time for settings to take effect.
+
+## Firebase Configuration
+
+The `Remote Config` contains key-value pairs where a single key stands for a feature name it controls and a value indicates whether this feature is enabled or not. You can create and manage the `Remote Config` parameters in the Firebase Console. Consider the following steps to open the required page:
+
+1. Open the [Firebase console](https://console.firebase.google.com/) and select your project.
+2. Click the `Remote Config` on the left panel.
+3. Fill the `Parameter key` and the `Default value` inputs.
+4. Click `Add parameter` button, to add the filled key-value pair.
+
+For the Metrics Web Application purposes, we should specify parameters for the `FPS Monitor`, `Login Form`, and `Renderer Display` features. Thus, the configurations may look like the following:
+
+```json
+isLoginFormEnabled: true
+isFpsMonitorEnabled: false
+isRendererDisplayEnabled: false
+```
+
+Once the configurations are ready, we should prepare the Metrics Web application to use them.
+
+## Metrics application
+
+The following paragraph provides an implementation of remote config integration for the Metrics Web Application by layers. Read more about layers and their responsibilities in the [Metrics Web Application architecture document](https://github.com/platform-platform/monorepo/blob/master/metrics/web/docs/01_metrics_web_application_architecture.md).
+
+### Data layer
+
+The data layer provides the `FirebaseRemoteConfigurationRepository` implementation that uses the [Firebase package](https://pub.dev/packages/firebase) available for Flutter for Web and the `RemoteConfigurationData` that represents the `RemoteConfiguration` entity.
+
+The following class diagram states the structure of the data layer:
+
+![Data layer diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/remote_config_design/metrics/web/docs/features/remote_config/diagrams/remote_config_data_layer_class_diagram.puml)
+
+### Domain layer
+
+The domain layer should provide an interface for the `FirebaseRemoteConfigurationRepository` we need to interact with the `Remote Config API`. Also, the layer provides all the use cases required to interact with the repository, and entities required for the `Remote Config` feature. Thus, the following list of classes should be implemented to fit the feature requirements:
+
+- Implement the `RemoteConfigurationRepository` interface with a `applyDefaults()`, `fetch()`, `activate()`, and `getConfiguration()` methods.
+- Add the `RemoteConfiguration` entity with fields that come from a remote API.
+- Add the `FetchRemoteConfigurationUseCase` to perform fetching configurations.
```

Comment:
Do we need the `SetDefaultRemoteConfigurationUseCase` as it seems to be a separate action for the application?